lib/cocoapods-downloader/git.rb

```diff
@@ -22,6 +22,23 @@ def checkout_options
         end
       end
 
+      def self.preprocess_options(options)
+        return options unless options[:branch]
+
+        command = ['ls-remote',
+                   options[:git],
+                   options[:branch]]
+        output = Executable.execute_command('git', command)
```

Comment:
those are instance methods, so they're not available here. it's a class method to avoid instancing downloaders just to preprocess options.